duckduckgo/Android
# Feature/feedback disambiguation

Task/Issue URL: https://app.asana.com/0/715106103902962/1109166030178654
iOS equivalent PR: https://github.com/duckduckgo/iOS/pull/420
Tech Design URL: 
CC

# Description
Adds a new flow for users to leave feedback from the `Settings` page.

This new flow tries to better capture the nature of the feedback by getting users to identify the category of the problem (is their feedback about missing features, bugs, search results etc...)

## Description of the flow

There is an initial happy / sad clarifier.
<img width="648" alt="Screenshot 2019-03-22 at 17 07 34" src="https://user-images.githubusercontent.com/1336281/54840496-166cad00-4cc5-11e9-968a-539bb8e2d2bd.png">


### Happy Flow

#### Rate, give more details or finish
In the happy flow, the users will be offered to chance to give us more info or to rate us, or to end the journey there.

They will only see this screen if they can rate us (app installed from Play Store) - otherwise this screen will be skipped and they'll be shown the "more details" page instead.

![device-2019-03-22-172037](https://user-images.githubusercontent.com/1336281/54841276-db6b7900-4cc6-11e9-89dc-2c78fe8e6d3d.png)

#### More details page
<img width="648" alt="Screenshot 2019-03-22 at 17 07 38" src="https://user-images.githubusercontent.com/1336281/54840651-5fbcfc80-4cc5-11e9-8ce4-004d356a2db2.png">

### Sad Flow

In this flow, the user will be shown a list of main reasons why they might be unhappy and asked to pick from the list.
<img width="648" alt="Screenshot 2019-03-22 at 17 07 46" src="https://user-images.githubusercontent.com/1336281/54840695-72373600-4cc5-11e9-8595-56c6a401fef9.png">

Depending on the option chosen, they'll either navigate to choosing a subreason, a page to provide "extra details", or they'll see a broken site page.

When choosing a subreason, they'll always end with navigating to the open ended "share details" page.

#### Missing Browser features
This will show subcategories. Choosing a subcategory will navigate to the open ended "share details" page.

#### Website loading issues
This will show a new broken site page (note, the existing "report broken site" screen is left as is)

#### Search results not good enough
This will show show subcategories. Choosing a subcategory will navigate to the open ended "share details" page.

#### Customization 
This will show subcategories. Choosing a subcategory will navigate to the open ended "share details" page.

#### None of these
This will jump straight to the open ended "share details" page.

**Steps to test this PR**:
1. Test all the flows, for each path observing pixel data and and the HTTP POST network request that happens when (if) you reach the terminal journey in that flow
1. Repeat with other (dark/light) theme

See the internal ticket for where details on where to look to verify pixel and POST data.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
